AsomamecoApi/test/Controllers/RolesControllerTests.cs

```diff
@@ -0,0 +1,265 @@
+            var result = await _controller.GetRoleById(id).ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Should().BeAssignableTo<ActionResult<Event>>();
+            result.Result.Should().BeOfType<OkObjectResult>();
+            result.Result.As<OkObjectResult>().Value
+                .Should()
+                .NotBeNull()
+                .And.BeOfType(roleMock.GetType());
+            _serviceMock.Verify(service => service.GetByID(id), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetByID_ShouldReturnNoContent_WhenNoDataFound()
+        {
+            // Arrange
+            Role response = null;
+            var id = _fixture.Create<int>();
+            _serviceMock.Setup(service => service.GetByID(id)).ReturnsAsync(response);
+
+            // Act
+            var result = await _controller.GetRoleById(id).ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Result.Should().BeOfType<NoContentResult>();
+            _serviceMock.Verify(service => service.GetByID(id), Times.Once);
+        }
+
+        [Fact]
+        public async Task GetByID_ShouldReturnBadRequest_WhenInputIsZero()
+        {
+            // Arrange
+            var id = 0;
+
+            // Act
+            var result = await _controller.GetRoleById(id).ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Result.Should().BeOfType<BadRequestObjectResult>();
+            _serviceMock.Verify(service => service.GetByID(id), Times.Never);
+        }
+
+        [Fact]
+        public async Task GetByID_ShouldReturnBadRequest_WhenInputIsLessThanZero()
+        {
+            // Arrange
+            var id = -1;
+
+            // Act
+            var result = await _controller.GetRoleById(id).ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Result.Should().BeOfType<BadRequestObjectResult>();
+            _serviceMock.Verify(service => service.GetByID(id), Times.Never);
+        }
+
+        //test exception
+        [Fact]
+        public async Task GetByID_ShouldReturnInternalServerError_WhenExceptionThrown()
+        {
+            // Arrange
+            var id = _fixture.Create<int>();
+            _serviceMock.Setup(service => service.GetByID(id)).ThrowsAsync(new Exception("Some error"));
+
+            // Act
+            var result = await _controller.GetRoleById(id).ConfigureAwait(false);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Result.Should().BeOfType<ObjectResult>();
+            result.Result.As<ObjectResult>().StatusCode.Should().Be(StatusCodes.Status500InternalServerError);
+            _serviceMock.Verify(service => service.GetByID(id), Times.Once);
+        }
+
+        [Fact]
+        public async Task Create_ShouldReturnCreatedResponse_WhenValidInput()
```

Comment:
Este test falla: Message: 
  Expected result to be assignable to Microsoft.AspNetCore.Mvc.ActionResult`1[src.Models.Event], but Microsoft.AspNetCore.Mvc.ActionResult`1[src.Models.Role] is not.